unstructured_ingest/v2/processes/connectors/gitlab.py

```diff
@@ -0,0 +1,304 @@
+
+        try:
+            gitlab = self.connection_config.get_client()
+            if self.connection_config.access_config.get_secret_value().access_token is not None:
+                gitlab.auth()
+            else:
+                gitlab.projects.get(self.connection_config.repo_path)
+
+        except GitlabError as gitlab_error:
+            logger.error(f"Failed to validate connection: {gitlab_error}", exc_info=True)
+            raise SourceConnectionError(f"Failed to validate connection: {gitlab_error}")
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        """Iterates over the GitLab repository tree and yields file metadata as `FileData` objects.
+
+        This method fetches the repository tree for the specified branch and iterates
+        over its contents. For each file (blob), it generates a `FileData` object containing
+        the file's metadata, path, and permissions.
+
+        Args:
+            **kwargs (Any): Additional keyword arguments (if required).
+
+        Yields:
+            FileData: A generator that yields `FileData` objects representing each file (blob)
+            in the repository.
+        """
+        project = self.connection_config.get_project()
+
+        ref = self.connection_config.git_branch or project.default_branch
+
+        git_tree = project.repository_tree(
+            ref=ref,
+            recursive=True,
+            iterator=True,
+            all=True,
+        )
+
+        for element in git_tree:
+            rel_path = element["path"].replace(self.connection_config.repo_path, "").lstrip("/")
+            if element["type"] == "blob":
+                record_locator = {
+                    "file_path": element["path"],
+                    "file_name": element["name"],
+                    "file_type": element["type"],
+                    "repo_path": self.connection_config.repo_path,
+                }
+                if self.connection_config.git_branch is not None:
+                    record_locator["branch"] = self.connection_config.git_branch
+
+                yield FileData(
+                    identifier=element["id"],
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element["path"],
+                        filename=element["path"].split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element["id"],
+                        record_locator=record_locator,
+                        permissions_data=[{"mode": element["mode"]}],
+                    ),
+                    additional_metadata={},
+                )
+
+
+class GitLabDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitLabDownloader(Downloader):
+    connection_config: GitLabConnectionConfig
+    download_config: GitLabDownloaderConfig
```

Comment:
Need this for the system to know to use the async implementation
```suggestion
    download_config: GitLabDownloaderConfig
    
        def is_async(self) -> bool:
        return True
```

Reply:
Hey Roman! This connector does not support `async `(only GitHub does). Because of that, I defined the method as run and omitted `is_async`